src/overlays/actors/ovl_En_Dekubaba/z_en_dekubaba.c

```diff
@@ -121,85 +133,1191 @@ static ColliderJntSphInit D_809E8FC8 = {
+    /* Master jump   */ DMG_ENTRY(4, DEKUBABA_DMGEFF_SWORD),
+    /* Unknown 1     */ DMG_ENTRY(0, DEKUBABA_DMGEFF_NONE),
+    /* Unblockable   */ DMG_ENTRY(0, DEKUBABA_DMGEFF_NONE),
+    /* Hammer jump   */ DMG_ENTRY(4, DEKUBABA_DMGEFF_NONE),
+    /* Unknown 2     */ DMG_ENTRY(0, DEKUBABA_DMGEFF_NONE),
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(targetArrowOffset, 1500, ICHAIN_STOP),
+};
+
+void EnDekubaba_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDekubaba* this = THIS;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 22.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &gDekuBabaSkel, &gDekuBabaFastChompAnim, this->jointTable,
+                   this->morphTable, 8);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->colliderElements);
+
+    if (this->actor.params == DEKUBABA_BIG) {
+        this->size = 2.5f;
+
+        for (i = 0; i < sJntSphInit.count; i++) {
+            this->collider.elements[i].dim.worldSphere.radius = this->collider.elements[i].dim.modelSphere.radius =
+                (sJntSphElementsInit[i].dim.modelSphere.radius * 2.50f);
+        }
+
+        // This and its counterpart below mean that a Deku Stick jumpslash will not trigger the Deku Stick drop route.
+        // (Of course they reckoned without each age being able to use the other's items, so Stick and Master Sword
+        // jumpslash can give the Stick drop as adult, and neither will as child.)
+        if (LINK_IS_CHILD) {
+            sBigDekuBabaDamageTable.table[0x1B] = DMG_ENTRY(4, DEKUBABA_DMGEFF_NONE); // DMG_JUMP_MASTER
+        }
+
+        CollisionCheck_SetInfo(&this->actor.colChkInfo, &sBigDekuBabaDamageTable, &sColChkInfoInit);
+        this->actor.colChkInfo.health = 4;
+        this->actor.naviEnemyId = 8; // Big Deku Baba
+        this->actor.targetMode = 2;
+    } else {
+        this->size = 1.0f;
+
+        for (i = 0; i < sJntSphInit.count; i++) {
+            this->collider.elements[i].dim.worldSphere.radius = this->collider.elements[i].dim.modelSphere.radius;
+        }
+
+        if (LINK_IS_CHILD) {
+            sDekuBabaDamageTable.table[0x1B] = DMG_ENTRY(4, DEKUBABA_DMGEFF_NONE); // DMG_JUMP_MASTER
+        }
+
+        CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDekuBabaDamageTable, &sColChkInfoInit);
+        this->actor.naviEnemyId = 7; // Deku Baba
+        this->actor.targetMode = 1;
+    }
+
+    EnDekubaba_SetupWait(this);
+    this->timer = 0;
+    this->boundFloor = NULL;
+    this->bodyPartsPos[3] = this->actor.home.pos;
+}
+
+void EnDekubaba_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDekubaba* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void EnDekubaba_DisableHitboxes(EnDekubaba* this) {
+    s32 i;
+
+    for (i = 1; i < 7; i++) {
```

Comment:
```suggestion
    for (i = 1; i < ARRAY_COUNT(this->colliderElements); i++) {
```
also for the other loops using `this->collider.elements[i]`